packages/devextreme/js/__internal/core/r1/dom_component_wrapper.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {
+  hasTemplate, InfernoComponent, InfernoEffect, renderTemplate,
+} from '@devextreme/runtime/inferno';
+import type { ComponentClass } from '@js/core/dom_component';
+import type { EventCallback } from '@js/renovation/ui/common/event_callback';
+import { getUpdatedOptions } from '@js/renovation/ui/common/utils/get_updated_options';
+import type { DisposeEffectReturn } from '@js/renovation/utils/effect_return';
+import type { RefObject } from 'inferno';
+import { createRef, normalizeProps } from 'inferno';
+
+import type DomComponent from '../../../core/dom_component';
+import { extend } from '../../../core/utils/extend';
+import { ConfigContext } from '../../../renovation/common/config_context';
+
+interface ComponentProps {
+  className?: string;
+  itemTemplate?: string | (() => string | HTMLElement);
+  valueChange?: EventCallback<any>;
+}
+
+export interface DomComponentWrapperProps {
+  componentType: ComponentClass<any>;
+  templateNames: string[];
+  componentProps: ComponentProps;
+}
+
+export class DomComponentWrapper extends InfernoComponent<DomComponentWrapperProps> {
+  public state: any = {};
+
+  public refs: any = null;
+
+  private readonly widgetRef: RefObject<HTMLDivElement> = createRef();
+
+  private instance: DomComponent | null = null;
+
+  private prevProps: ComponentProps | null = null;
+
+  constructor(props) {
+    super(props);
+    this.getInstance = this.getInstance.bind(this);
+    this.setupWidget = this.setupWidget.bind(this);
+    this.updateWidget = this.updateWidget.bind(this);
+  }
+
+  getConfig(): any {
+    const { id } = (ConfigContext as any);
+    if (this.context[id]) {
+      return this.context[id];
+    }
+    return (ConfigContext as any).defaultValue;
+  }
+
+  render(): JSX.Element {
+    return (
+      <div
+        ref={this.widgetRef}
+        className={this.props.componentProps.className}
+        {...this.props}
+      />
+    );
+  }
+
+  componentWillUpdate(nextProps: DomComponentWrapperProps, nextState, context): void {
+    super.componentWillUpdate(nextProps, nextState, context);
+  }
```

Comment:
```suggestion
```

looks like we can safely delete it

Reply:
Unfortunately not( It was added [here](https://github.com/DevExpress/DevExtreme/pull/27822/commits/1fcb92f7d73344966889989b4baf666438ed7792) because the component had linter issues.

Reply:
Oh, yep, type clashes between React and Inferno...
Yep, let's leave it now